### ISSUE TITLE
No intellisense when exporting es3 style class 

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#8144]
_From @DenysYefimenko on April 18, 2016 11:9_

- VSCode Version: 1.0.0
- OS Version: Windows 10 Enterprise

Steps to Reproduce:

1. Create two js files.

**moduleA.js**
```javascript
var constructorFunction = function () {};
constructorFunction.prototype.prototypeMethod = function () {};
module.exports = constructorFunction;
```
**moduleB.js**
```javascript
var moduleA = require('./moduleA');
var constructedObject = new moduleA();
constructedObject.
```
2. Try get intellisense suggestion after dot in **moduleB**. 
3. No intellisense provided.

4. If I change code in **moduleA** as following:
```javascript
function constructorFunction () {}
constructorFunction.prototype.prototypeMethod = function () {};
module.exports = constructorFunction;
```
intellisense in **moduleB** starts working.

Expected result: intellisense works no matter how exported function is defined.

_Copied from original issue: Microsoft/vscode#5442_

**Comment by dbaeumer** [comment]
@DenysYefimenko the change part reads exactly the same as in steps. e.g 
```js
function constructorFunction () {} 
 constructorFunction.prototype.prototypeMethod = function () {}; 
 module.exports = constructorFunction; 
```

Are we missing something ?

**Comment by dbaeumer** [comment]
_From @DenysYefimenko on April 18, 2016 11:58_

Yes, you do :) In changed part I declared constructorFunction without using `var variable = function() {...}` construction.

**Comment by dbaeumer** [comment]
Thanks. Missed that.

Moving to TS.

**Comment by billti** [comment]
Interesting... ping @RyanCavanaugh . Do we infer JS types from function declarations differently to function expressions?

**Comment by sandersn** [comment]
Looks related (or identical) to #7586, where the language service treats TypeScript *and* JavaScript function expressions worse than function declarations. I think that fixing #7586 will fix this, but this example should be added as a test.

**Comment by sandersn** [comment]
Actually, the fix for #7586 (#9010) doesn't fix this. Looks like it will need a separate fix.

**Comment by sandersn** [comment]
Fix is up at #9088.
